[CCL]CheckTensorsInDifferentDevices存在同名重复定义，导致编译失败

### bug描述 Describe the Bug

版本 develop 76b328bcb687dc91886aba878711cd9bbc2c60ed
出问题的PR 四天前 https://github.com/PaddlePaddle/Paddle/pull/56301
编译参数
```
cmake .. -DWITH_CUSTOM_DEVICE=ON -DWITH_TENSORRT=ON
```


编译出错
```
/usr/bin/ld: ../distributed/collective/libprocess_group_custom.a(process_group_custom.cc.o): in function `paddle::distributed::CheckTensorsInDifferentDevices(std::vector<phi::DenseTensor, std::allocator<phi::DenseTensor> > const&, unsigned long)':
process_group_custom.cc:(.text+0x3160): multiple definition of `paddle::distributed::CheckTensorsInDifferentDevices(std::vector<phi::DenseTensor, std::allocator<phi::DenseTensor> > const&, unsigned long)'; ../distributed/collective/libprocess_group_nccl.a(process_group_nccl.cc.o):process_group_nccl.cc:(.text+0x3bd0): first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [paddle/fluid/pybind/CMakeFiles/kernel_signature_generator.dir/build.make:1127: paddle/fluid/pybind/kernel_signature_generator] Error 1
make[1]: *** [CMakeFiles/Makefile2:93033: paddle/fluid/pybind/CMakeFiles/kernel_signature_generator.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
/usr/bin/ld: ../../distributed/collective/libprocess_group_custom.a(process_group_custom.cc.o): in function `paddle::distributed::CheckTensorsInDifferentDevices(std::vector<phi::DenseTensor, std::allocator<phi::DenseTensor> > const&, unsigned long)':
process_group_custom.cc:(.text+0x3160): multiple definition of `paddle::distributed::CheckTensorsInDifferentDevices(std::vector<phi::DenseTensor, std::allocator<phi::DenseTensor> > const&, unsigned long)'; ../../distributed/collective/libprocess_group_nccl.a(process_group_nccl.cc.o):process_group_nccl.cc:(.text+0x3bd0): first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [paddle/fluid/eager/auto_code_generator/CMakeFiles/eager_generator.dir/build.make:1124: paddle/fluid/eager/auto_code_generator/eager_generator] Error 1
make[1]: *** [CMakeFiles/Makefile2:94786: paddle/fluid/eager/auto_code_generator/CMakeFiles/eager_generator.dir/all] Error 2
/usr/bin/ld: ../distributed/collective/libprocess_group_custom.a(process_group_custom.cc.o): in function `paddle::distributed::CheckTensorsInDifferentDevices(std::vector<phi::DenseTensor, std::allocator<phi::DenseTensor> > const&, unsigned long)':
process_group_custom.cc:(.text+0x3160): multiple definition of `paddle::distributed::CheckTensorsInDifferentDevices(std::vector<phi::DenseTensor, std::allocator<phi::DenseTensor> > const&, unsigned long)'; ../distributed/collective/libprocess_group_nccl.a(process_group_nccl.cc.o):process_group_nccl.cc:(.text+0x3bd0): first defined here
collect2: error: ld returned 1 exit status
make[2]: *** [paddle/fluid/pybind/CMakeFiles/eager_legacy_op_function_generator.dir/build.make:1127: paddle/fluid/pybind/eager_legacy_op_function_generator] Error 1
make[1]: *** [CMakeFiles/Makefile2:92412: paddle/fluid/pybind/CMakeFiles/eager_legacy_op_function_generator.dir/all] Error 2

```

原因 
CheckTensorsInDifferentDevices 存在重复定义
paddle/fluid/distributed/collective/process_group_nccl.cc
paddle/fluid/distributed/collective/process_group_custom.cc

CheckTensorsInDifferentDevices 是在paddle::distributed 下的全局函数



### 其他补充信息 Additional Supplementary Information

_No response_

--- a/paddle/fluid/distributed/collective/process_group_custom.cc
+++ b/paddle/fluid/distributed/collective/process_group_custom.cc
@@ -815,7 +815,7 @@ std::shared_ptr<ProcessGroup::Task> ProcessGroupCustom::Broadcast(
       CommType::BROADCAST);
 }
 
-void CheckTensorsInDifferentDevices(
+inline void CheckTensorsInDifferentDevices(
     const std::vector<phi::DenseTensor>& tensors, const size_t num_devices) {
   PADDLE_ENFORCE_EQ(
       tensors.empty(),

